ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/ip_security.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.plugin_utils.strip_empties import strip_null_from_data
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+from .utils import UtilsMixin
+
+
+class IpSecurityMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def ip_security(self) -> dict | None:
+        """
+        ip_security set based on wan_ipsec_profiles data_model
+
+        If `data_plane` is not configured, `control_plane` data is used for both
+        Data Plane and Control Plane.
+        """
+        if not self.shared_utils.wan_role:
+            return None
+
+        wan_ipsec_profiles = get(self._hostvars, "wan_ipsec_profiles", required=True)
+
+        # Structure initialization
+        ip_security = {"ike_policies": [], "sa_policies": [], "profiles": []}
+
+        if (data_plane := get(wan_ipsec_profiles, "data_plane", None)) is not None:
+            self._append_data_plane(ip_security, data_plane)
+        if (control_plane := get(wan_ipsec_profiles, "control_plane", None)) is not None:
+            self._append_control_plane(ip_security, control_plane)
+
+        return strip_null_from_data(ip_security)
+
+    def _append_data_plane(self, ip_security: dict, data_plane_config: dict) -> None:
+        """
+        In place update of ip_security
+        """
+        ike_policy_name = get(data_plane_config, "ike_policy_name", "DP-IKE-POLICY")
```

Comment:
I know it works the same, but just for clarity:
```suggestion
        ike_policy_name = get(data_plane_config, "ike_policy_name", default="DP-IKE-POLICY")
```